### PR TITLE
IGNITE-12828 Fixes NPE during CQ registration with failed deployment.

### PR DESCRIPTION
**Pull request by ololo3000**

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/query/continuous/CacheContinuousQueryFilterDeploymentFailedTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+        grid(0).createCache(new CacheConfiguration<>(DEFAULT_CACHE_NAME));
+
+        ContinuousQuery<Integer, Integer> qry = new ContinuousQuery<>();
+
+        Class<Factory<CacheEntryEventFilter<Integer, Integer>>> remoteFilterFactoryCls =
+            (Class<Factory<CacheEntryEventFilter<Integer, Integer>>>)getExternalClassLoader()
+                .loadClass(EXT_FILTER_FACTORY_CLS);
+
+        Factory<CacheEntryEventFilter<Integer, Integer>> remoteFilterFactory = remoteFilterFactoryCls.newInstance();
```
Comment (NSAmelchev) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/query/continuous/CacheContinuousQueryFilterDeploymentFailedTest.java:72]: Let's use a shortened form `rmt`

Comment (ololo3000) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/query/continuous/CacheContinuousQueryFilterDeploymentFailedTest.java]: Thanks. Done.

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/query/continuous/CacheContinuousQueryFilterDeploymentFailedTest.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            return locQrys.size() == 0;
+        }));
+
+    }
+
+    /**
+     * Wrapper for {@link DiscoverySpiListener} with ability to update specified counter for every handled
+     * {@link StopRoutineDiscoveryMessage}.
+     */
+    private static class TestDiscoverySpiListener implements DiscoverySpiListener {
```
Comment (NSAmelchev) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/query/continuous/CacheContinuousQueryFilterDeploymentFailedTest.java:142]: `GridTestUtils.DiscoverySpiListenerWrapper` can be reused

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/query/continuous/CacheContinuousQueryFilterDeploymentFailedTest.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+            Collection<ClusterNode> topSnapshot,
+            @Nullable Map<Long, Collection<ClusterNode>> topHist,
+            @Nullable DiscoverySpiCustomMessage data
+        ) {
+            IgniteFuture<?> fut = lsnr.onDiscovery(type, topVer, node, topSnapshot, topHist, data);
+
+            DiscoveryCustomMessage customMsg = data == null ?
+                null : (DiscoveryCustomMessage)U.field(data, "delegate");
+
+            if (customMsg instanceof StopRoutineDiscoveryMessage)
```
Comment (NSAmelchev) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/query/continuous/CacheContinuousQueryFilterDeploymentFailedTest.java:174]: Maybe ack message (`StopRoutineAckDiscoveryMessage`) should be checked?

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/query/continuous/CacheContinuousQueryFilterDeploymentFailedTest.java [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        List<Ignite> grids = G.allGrids();
+
+        GridTestUtils.waitForCondition(() -> stopRoutineMsgCntr.get() == grids.size(), getTestTimeout());
+
+        assertTrue(grids.stream().allMatch(ignite -> {
+            SystemView<ContinuousQueryView> locQrys = ((IgniteEx)ignite).context().systemView().view(CQ_SYS_VIEW);
+
+            return locQrys.size() == 0;
+        }));
+
```
Comment (NSAmelchev) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/query/continuous/CacheContinuousQueryFilterDeploymentFailedTest.java:135]: Unnecessary line break

Comment (ololo3000) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/cache/query/continuous/CacheContinuousQueryFilterDeploymentFailedTest.java]: Thanks. Done.

**Pull request merged**